src/routes/github/github-oauth-router.ts

```diff
@@ -8,17 +8,20 @@ import { Tracer } from "config/tracer";
 import { envVars } from "config/env";
 import { GithubAPI } from "config/github-api";
 import { Errors } from "config/errors";
-import { getGitHubHostname, getGitHubApiUrl } from "~/src/util/get-github-client-config";
+import { getGitHubApiUrl } from "~/src/util/get-github-client-config";
 import { createHashWithSharedSecret } from "utils/encryption";
 
 const logger = getLogger("github-oauth");
-
-const githubClient = envVars.GITHUB_CLIENT_ID;
-const githubSecret = envVars.GITHUB_CLIENT_SECRET;
 const baseURL = envVars.APP_URL;
 const scopes = ["user", "repo"];
 const callbackPath = "/callback";
 
+const getRedirectUrl = async (req, res, state) => {
+	const { hostname, clientId } = res.locals.gitHubAppConfig;
+	const callbackURI = new URL(`${req.baseUrl}${req.path}/..${callbackPath}`, baseURL).toString();
```

Comment:
There must be a better way to construct this callbackURI right? 
Like `${baseURL}/github/${res.locals.gitHubAppConfig.uuid}/callback` for GHES?
and `${baseURL}/github/callback` for cloud?

Reply:
that `..` in  `..${callbackPath}` feels weird.

Reply:
Updated to simple query concatenation `${appUrl}${baseUrl}${callbackPath}`